sync/awsdatabricks.py

```diff
@@ -623,16 +703,71 @@ def _get_task_cluster(task: dict, clusters: list) -> Response[dict]:
     return Response(result=cluster)
 
 
-def _get_eventlog(task: dict, cluster: dict) -> Response[bytes]:
+def _s3_contents_have_all_rollover_logs(contents: list[dict], run_end_time_seconds: float):
+    final_rollover_log = contents and next(
+        (
+            content
+            for content in contents
+            if Path(content["Key"]).stem in {"eventlog", "eventlog.gz", "eventlog.json.gz"}
+        ),
+        False,
+    )
+    return (
+        # Related to - https://docs.databricks.com/clusters/configure.html#cluster-log-delivery-1
+        # We want to make sure that the final_rollover_log has had data written to it AFTER the run has actually
+        # completed. We use this as a signal that Databricks has flushed all event log data to S3, and that we
+        # can proceed with the upload + prediction. If this proves to be finnicky in some way, we can always
+        # wait for the full 5 minutes after the run_end_time_seconds
+        final_rollover_log
+        and final_rollover_log["LastModified"].timestamp() >= run_end_time_seconds
+    )
+
+
+def event_log_poll_duration_seconds():
+    """Convenience function to aid testing"""
+    return 15
+
+
+def _get_eventlog(task: dict, cluster: dict, run_end_time_millis: int) -> Response[bytes]:
     log_url = cluster.get("cluster_log_conf", {}).get("s3", {}).get("destination")
     if log_url:
         parsed_log_url = urlparse(log_url)
-
+        cluster_id = task.get("cluster_instance", {}).get("cluster_id")
         s3 = boto.client("s3")
-        contents = s3.list_objects_v2(
-            Bucket=parsed_log_url.netloc,
-            Prefix=f"{parsed_log_url.path.strip('/')}/{task.get('cluster_instance', {}).get('cluster_id')}/eventlog/{task.get('cluster_instance').get('cluster_id')}",
-        ).get("Contents")
+
+        if stripped_path := parsed_log_url.path.strip("/"):
+            prefix = f"{stripped_path}/{cluster_id}/eventlog/{cluster_id}"
+        else:
+            # If the event log destination is just a *bucket* without any sub-path, then we don't want to include
+            #  a leading `/` in our Prefix (which will make it so that we never actually find the event log)
+            prefix = f"{cluster_id}/eventlog/{cluster_id}"
```

Comment:
I don't think that will work, because if `stripped_path == ""`, then we are pre-prending a `/` to our `Prefix`, which is what causes the bug. If the log path is _just_ a bucket, then we don't want any leading `/` in there at all. I love getting rid of `if` statements, but I don't see a great way to for this one...

Reply:
That's the reason for the strip at the end of the statement:
```
>>> url = "s3://megacorp-jobs"
>>> cluster_id = "0411-011041-ldaw3bkd"
>>> stripped_path = urlparse(url).path
>>> f"{stripped_path}/{cluster_id}/eventlog/{cluster_id}".strip("/")
'0411-011041-ldaw3bkd/eventlog/0411-011041-ldaw3bkd'
>>> url = "s3://megacorp-jobs/projects"
>>> stripped_path = urlparse(url).path
>>> f"{stripped_path}/{cluster_id}/eventlog/{cluster_id}".strip("/")
'projects/0411-011041-ldaw3bkd/eventlog/0411-011041-ldaw3bkd'
```